ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,55 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                context("message cannot be decoded or decrypted") {
+
+                    // RTL7e
+                    it("should deliver with encoding attribute set indicating the residual encoding and error should be emitted") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channelOptions = ARTChannelOptions(encrypted: ARTCrypto.defaultParams())
```

Comment:
@mattheworiordan Sorry, you're right, Crypto is public. But TZ3b says that the key parameter is optional, and TZ3a says that if the key isn't passed it is generated randomly. Thus the only thing that should be changed here is the ARTChannelOptions constructor. (I'll submit a PR with that.)

By the way, and this is kind of off-topic here, but does it make sense how we're using the IV? The specified API doesn't even let you pass a IV to `getCipherParams`, it just generates one randomly. But AFAICT you won't be able to decrypt those messages unless you have the same IV on the receiving end. Besides, we're CBC-chaining _all_ the messages in a channel, so the receiving end will need to get _all_ the messages published in that channel, and if it misses any of them because it subscribes later or there's a suspension or something.

Not saying that it's wrong, and I don't immediately see a better way to do it, just want to confirm that I'm getting it right.